src/iter.rs

```diff
@@ -57,11 +57,97 @@ where
     }
 }
 
+#[derive(Debug)]
```

Comment:
This new code should go somewhere nearer the bottom of the module, since the project is organized in a top-down fashion.